### PR TITLE
feat(coderd): store workspace proxy version in the database

### PR DESCRIPTION
**Pull request by johnstcn**
Part of https://github.com/coder/coder/issues/9558

When workspace proxies come online, they hit the `/register` endpoint with a payload.
Part of this payload is the proxy version (`buildinfo.Version`).
Right now the `/register` handler performs a check that the major and minor versions match before allowing the registration to complete.

This change just adds a database column and updates the stored version with the value reported by workspace proxies.

### CODE REVIEW

File: coderd/database/migrations/000170_workspaceproxy_version.down.sql
Comment (johnstcn) [inline coderd/database/migrations/000170_workspaceproxy_version.down.sql:1]: There's a 169 in the pipeline already, so renaming preemptively

**Review by mtojek — APPROVED** [review]
In case of future improvements, we don't expect another `version` column? Let's say, protocol version, agent version, foobar version, etc.

**Comment by johnstcn** [comment]
> In case of future improvements, we don't expect another `version` column? Let's say, protocol version, agent version, foobar version, etc.

`workspace_proxies.proxy_version` is really cumbersome. I initially considered `api_version` but felt that would be easily confused with the corresponding `workspace_agents.api_version`.

**Comment by mtojek** [comment]
`release_version`, `app_version` ?

**Review by Emyrk — DISMISSED** [review]
We need the version field on the codersdk type too right?

https://github.com/coder/coder/blob/7c4043d9adf043d233c76837d65fcad14456bbc3/codersdk/workspaceproxy.go#L61-L61

**Comment by Emyrk** [comment]
And the convert https://github.com/coder/coder/blob/7c4043d9adf043d233c76837d65fcad14456bbc3/enterprise/coderd/workspaceproxy.go#L948-L948

**Review by Emyrk — APPROVED** [review]

**Pull request merged**